src/App.js

```diff
@@ -338,7 +344,14 @@ class _AppStackWrapper extends React.Component {
         screen = <RecoverPin navigation={this.props.navigation} />;
       }
 
-      if (this.props.walletStartState === WALLET_STATUS.LOADING || this.props.isScreenLocked) {
+      const currentRouteIdx = this.props.navigation.state.index;
+      const currentRoute = this.props.navigation.state.routes[currentRouteIdx];
+
+      // We need to hide the loading screen if the wallet is being reset, otherwise
+      // the loading screen will be on top of the reset screen.
+      if (currentRoute.routeName !== 'ResetWallet'
```

Comment:
I don't understand the need for this. The wallet is never loading because we move from the locked screen to the reset screen, so why it enters this `if (this.props.walletStartState === WALLET_STATUS.LOADING ..` in the first place?

Reply:
We need this because, currently, the wallet defaults to `LOADING`, so when the `PinScreen` hides, it automagically navigates to `LoadHistoryScreen` which in turn will dispatch `startWalletRequested`